### PR TITLE
feat: add badge php version require

### LINKED ISSUE #607: Add php version badge (and maybe other dependencies)
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | yes
| BC Break    | no

#### Summary

According to the tests fixtures
https://raw.githubusercontent.com/PUGX/badge-poser/f71d0b72d22f481fdccf52f426eba3a59831b189/tests/Badge/Model/fixtures/packages/imagine-bundle.json
when the call is made to packagist, the `require` section is provided.

So a badge
```
php: >=5.3.2
```
could be possible

This would allow people to know directly in the Readme the compatible version without looking at the composer.json. (and it will be always up to date with the last version)

I think about this especially for the php version but there might be some interest for some other dependencies version...

### PR DESCRIPTION
**Pull request by JellyBellyDev**
closes #607

WDYT?

Schermata 2021-08-24 alle 17 25 43

The new badge works with all requires, example:
```
/phpunit/phpunit/require/php
/phpunit/phpunit/require/ext-gd
/phpunit/phpunit/require/badges/poser
```

### CODE REVIEW

**Review by AlessandroMinoccheri — APPROVED** [review]
LGTM 👍 
Amazing!

File: src/Badge/Model/UseCase/CreateRequireBadge.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    protected function prepareText(Package $package, ?string $context): string
+    {
+        $require = $package->getLatestRequire((string) $context);
+
+        if (empty($require)) {
+            return self::TEXT_NO_REQUIRE;
+        }
+
+        return $require;
```
Comment (garak) [inline src/Badge/Model/UseCase/CreateRequireBadge.php:56]: Can we use `$require ?? self::TEXT_NO_REQUIRE`?

**Review by garak — CHANGES_REQUESTED** [review]

File: src/Badge/Model/UseCase/CreateRequireBadge.php (hunk 1 above)
Comment (JellyBellyDev) [reply src/Badge/Model/UseCase/CreateRequireBadge.php:56]: nope! :(

```php
$foo = '';
$bar = $foo ?? 'default';
echo $bar;
```

File: src/Badge/Model/UseCase/CreateRequireBadge.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Create the 'require' image with the standard Font and standard Image.
+ */
+final class CreateRequireBadge extends BaseCreatePackagistImage
+{
+    public const COLOR = '787CB5';
+    public const TEXT_NO_REQUIRE = ' - ';
+
+    private const TTL_DEFAULT_MAXAGE = CacheableBadge::TTL_ONE_HOUR;
```
Comment (fabiocicerchia) [inline src/Badge/Model/UseCase/CreateRequireBadge.php:26]: maybe we can even store the badge for longer, since it shouldn't change that often

**Review by fabiocicerchia — APPROVED** [review]

File: src/Badge/Model/UseCase/CreateRequireBadge.php (hunk 2 above)
Comment (JellyBellyDev) [reply src/Badge/Model/UseCase/CreateRequireBadge.php:26]: could change how could the version name change... the require are taken from the latest stable version.

File: src/Badge/Model/UseCase/CreateRequireBadge.php (hunk 1 above)
Comment (garak) [reply src/Badge/Model/UseCase/CreateRequireBadge.php:56]: Sorry for the late response.
Can't we at least use a ternary? E.g. `return empty($require) ? self::TEXT_NO_REQUIRE : $require;`

Comment (JellyBellyDev) [reply src/Badge/Model/UseCase/CreateRequireBadge.php]: Nice catch! Done! 👍

**Review by garak — APPROVED** [review]

**Comment by VincentLanglet** [comment]
I tried `https://poser.pugx.org/vincentlanglet/twig-cs-fixer/require/php` with no success.

What am I doing wrong ?

**Comment by fabiocicerchia** [comment]
@VincentLanglet We have just released the new version, you should be able now to see correctly the php badge.

**Pull request merged**